### ISSUE TITLE
JS Bundle Error after upgrading to 0.70.1.

### ISSUE THREAD

**Issue by SectionTN** [facebook/react-native#34712]
### New Version

0.70.1

### Old Version

0.69.5

### Build Target(s)

android debug and maybe release too

### Output of `react-native info`

```
info Fetching system and libraries information...
System:
    OS: Linux 5.19 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (4) x64 Intel(R) Core(TM) i7-2640M CPU @ 2.80GHz
    Memory: 1.05 GB / 3.72 GB
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.9.0 - /snap/bin/node
    Yarn: 1.22.19 - /snap/bin/yarn
    npm: 8.19.1 - /snap/bin/npm
    Watchman: Not Found
  SDKs:
    Android SDK:
      API Levels: 29, 30, 31, 32, 33
      Build Tools: 29.0.2, 30.0.3, 31.0.0, 32.0.0, 33.0.0
      Android NDK: Not Found
  IDEs:
    Android Studio: AI-212.5712.43.2112.8815526
  Languages:
    Java: 1.8.0_342 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0 
    react-native: 0.70.1 => 0.70.1 
  npmGlobalPackages:
    *react-native*: Not Found
```

### Issue and Reproduction Steps

My issue is no library has been detected after the upgrade, for example:
```
error: Error: Unable to resolve module react-native-onesignal from /home/section/Documents/GitHub/MoonMeet-CrossPlatform/index.js: react-native-onesignal could not be found within the project or in these directories:
  node_modules
  ../../../node_modules
   7 |  */
   8 | import {AppRegistry} from 'react-native';
>  9 | import OneSignal from 'react-native-onesignal';                                                                                                        
     |                        ^
  10 | import App from './App';
  11 | import {name as MoonMeet} from './app.json';
  12 | import {ONESIGNAL_APP_ID} from './src/secrets/sensitive';
    at ModuleResolver.resolveDependency (/home/section/Documents/GitHub/MoonMeet-CrossPlatform/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:178:15)
    at DependencyGraph.resolveDependency (/home/section/Documents/GitHub/MoonMeet-CrossPlatform/node_modules/metro/src/node-haste/DependencyGraph.js:264:43)
    at Object.resolve (/home/section/Documents/GitHub/MoonMeet-CrossPlatform/node_modules/metro/src/lib/transformHelpers.js:170:21)
    at resolveDependencies (/home/section/Documents/GitHub/MoonMeet-CrossPlatform/node_modules/metro/src/DeltaBundler/graphOperations.js:466:33)
    at processModule (/home/section/Documents/GitHub/MoonMeet-CrossPlatform/node_modules/metro/src/DeltaBundler/graphOperations.js:232:31)
    at async traverseDependenciesForSingleFile (/home/section/Documents/GitHub/MoonMeet-CrossPlatform/node_modules/metro/src/DeltaBundler/graphOperations.js:221:3)
    at async Promise.all (index 0)
    at async initialTraverseDependencies (/home/section/Documents/GitHub/MoonMeet-CrossPlatform/node_modules/metro/src/DeltaBundler/graphOperations.js:204:3)
    at async DeltaCalculator._getChangedDependencies (/home/section/Documents/GitHub/MoonMeet-CrossPlatform/node_modules/metro/src/DeltaBundler/DeltaCalculator.js:208:25)
    at async DeltaCalculator.getDelta (/home/section/Documents/GitHub/MoonMeet-CrossPlatform/node_modules/metro/src/DeltaBundler/DeltaCalculator.js:90:16)
```
if I comment (`//`) OneSignal and Notifee codes, I got ./app.json is not found :/ which is 100000% inside my root project folder.
I'm pretty sure it's a wrong node modules folder or something, when i rollback to 0.69.5 my application works fine

steps:
1. followed `react-native community upgrade helper` instruction strictly two times.
2. did `yarn start --reset-cache`
3. did `cd android && ./gradlew clean && cd ..`
4. did `yarn android`
5. building looks fine without any errors
6. after Gradle task `installDebug` 
7. got those errors above

**Comment by SectionTN** [comment]
any updates on this?
also why is the directory `../../../node_modules`, i think it's wrong.

**Comment by marceloch2** [comment]
I have the same for IOS.

**Comment by SectionTN** [comment]
i hope someone finds a solution.

**Comment by SectionTN** [comment]
@cortinico (sorry for the ping with your permission) could you check this issue please?

**Comment by cortinico** [comment]
> @cortinico (sorry for the ping with your permission) could you check this issue please?

Can you provide a reproducer?

**Comment by SectionTN** [comment]
I did init a new React Native Project 0.70.1, nothing went wrong with the bundler, but it's not working with my project, could you try to run this, [repo link](https://github.com/MoonMeet/MoonMeet-CrossPlatform/tree/upgrade/RN70.1), choose branch `upgrade/RN70.1`

**Comment by cortinico** [comment]
> I did init a new React Native Project 0.70.1, nothing went wrong with the bundler, but it's not working with my project, could you try to run this, [repo link](https://github.com/MoonMeet/MoonMeet-CrossPlatform/tree/upgrade/RN70.1), choose branch `upgrade/RN70.1`

I could not build your project @SectionTN. A simple `yarn && yarn android` fails with:

```
> Could not resolve all files for configuration ':react-native-clipboard_clipboard:classpath'.
   > Could not download builder-3.2.1.jar (com.android.tools.build:builder:3.2.1)
      > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/build/builder/3.2.1/builder-3.2.1.jar'.
```

As you're depending on `react-native-clipboard` which is on AGP 3 (which is quite of an old version) and can't be resolved anymore. Your project is also full of dependencies making hard to pinpoint the problem. Can you create a smaller reproducer?

**Comment by SectionTN** [comment]
> Yes my project depending on `react-native-clipboard/clipboard` package.
> i can build my project on linux / windows without problems, idk why it's showing you a build error there.
i have did `npx react-native init Project` and tried building, nothing went wrong on a fresh project, as you said it's hard to find the problem here, I'll add any comment about may help resolving this problem if possible.

**Comment by SectionTN** [comment]
re-upgraded again and encountered the same issue ![Screenshot_20220921-175922_Moon Meet.png](https://user-images.githubusercontent.com/73752904/191566486-1070ae94-d714-42fb-8eca-f3210321e4ed.png)

**Comment by AliMamed** [comment]
For me the problem was in `metro.config.js`.
<img width="558" alt="Снимок экрана 2022-09-27 в 13 45 01" src="https://user-images.githubusercontent.com/19426160/192506236-7af33cbc-ee59-47da-ba36-d0e40016a2d7.png">
Lines 9-11 caused the problem.
```
  resolver: {
    sourceExts: ['jsx', 'js', 'ts', 'tsx'],
  },
```
Adding `json` extension to the array solved the problem. I must admit that the error reporting for me was also confusing: I had the errors of resolving other modules rather than anything wrong with `app.json`.

**Comment by SectionTN** [comment]
![Screenshot_20220927-121223_GitHub.png](https://user-images.githubusercontent.com/73752904/192510904-7074ea9b-628b-4f7e-8121-ff6db520e806.png)
already have this config inside my metro config

**Comment by AliMamed** [comment]
> already have this config inside my metro config
so add `json` extension there

**Comment by SectionTN** [comment]
thanks adding `json` fixed my problem

**Comment by princefishthrower** [comment]
Amazing... this worked for me too!!!!!!! But what is actually behind the need for `json`? Very weird in my opinion that this should affect importing modules in `index.js`...

**Comment by SectionTN** [comment]
i think `json` is needed for app.json to register the app or something like that.

**Comment by AhmedAli125** [comment]
> For me the problem was in `metro.config.js`. <img alt="Снимок экрана 2022-09-27 в 13 45 01" width="558" src="https://user-images.githubusercontent.com/19426160/192506236-7af33cbc-ee59-47da-ba36-d0e40016a2d7.png"> Lines 9-11 caused the problem.
> 
> ```
>   resolver: {
>     sourceExts: ['jsx', 'js', 'ts', 'tsx'],
>   },
> ```
> 
> Adding `json` extension to the array solved the problem. I must admit that the error reporting for me was also confusing: I had the errors of resolving other modules rather than anything wrong with `app.json`.
> 
> @SectionTN I see similar config in your repo, so I think you can try fixing your project by adding `json` to the array.


Thanks @AliMamed adding json extension in array also solved issue on my end.

**Comment by rafaelmaeuer** [comment]
I had a similar problem with `Unable to resolve module @notifee/react-native`.
Adding `json` to `metro.config.js` resolved the issue for me.

**Comment by Sagargajul41** [comment]
Hi everyone,
I am getting below error after my react native version moved from 0.64 to 0.70 in my current project.
New Version
0.70.13
Old Version
0.64.2
Can anyone help me with this issue?

error: Error: Unable to resolve module ./app.json from D:\VR-Setup\UI-Projects\newproject\index.js:

None of these files exist:

app.json(.native|.android.jsx|.native.jsx|.jsx|.android.js|.native.js|.js)
app.json\index(.native|.android.jsx|.native.jsx|.jsx|.android.js|.native.js|.js)
5 | import {AppRegistry} from 'react-native';
6 | import App from './App';
7 | import {name as appName} from './app.json';
| ^
8 | import 'react-native-gesture-handler';
9 | AppRegistry.registerComponent(appName, () => App);
10 |
at ModuleResolver.resolveDependency (D:\VR-Setup\UI-Projects\newproject\node_modules\metro\src\node-haste\DependencyGraph\ModuleResolution.js:152:15)
at DependencyGraph.resolveDependency (D:\VR-Setup\UI-Projects\newproject\node_modules\metro\src\node-haste\DependencyGraph.js:264:43)
at Object.resolve (D:\VR-Setup\UI-Projects\newproject\node_modules\metro\src\lib\transformHelpers.js:170:21)
at resolveDependencies (D:\VR-Setup\UI-Projects\newproject\node_modules\metro\src\DeltaBundler\graphOperations.js:466:33)
at processModule (D:\VR-Setup\UI-Projects\newproject\node_modules\metro\src\DeltaBundler\graphOperations.js:232:31) at runMicrotasks ()
at processTicksAndRejections (node:internal/process/task_queues:96:5)
at async traverseDependenciesForSingleFile (D:\VR-Setup\UI-Projects\newproject\node_modules\metro\src\DeltaBundler\graphOperations.js:221:3)
at async Promise.all (index 0)
at async initialTraverseDependencies (D:\VR-Setup\UI-Projects\newproject\node_modules\metro\src\DeltaBundler\graphOperations.js:204:3)

**Comment by Sagargajul41** [comment]
@all @AliMamed  If I made changes in metro.config.js file , 

I am getting below error , Can anyone help to fix this issue? ( It's been long time I am trying to resolve it )


 ERROR  Invariant Violation: ViewPropTypes has been removed from React Native. Migrate to ViewPropTypes exported from 'deprecated-react-native-prop-types'., js engine: hermes
 ERROR  Invariant Violation: Failed to call into JavaScript module method AppRegistry.runApplication(). Module has not been registered as callable. Registered callable JavaScript modules (n = 11): Systrace, JSTimers, HeapCapture, SamplingProfiler, RCTLog, RCTDeviceEventEmitter, RCTNativeAppEventEmitter, GlobalPerformanceLogger, JSDevSupportModule, HMRClient, RCTEventEmitter.
        A frequent cause of the error is that the application entry file path is incorrect. This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native., js engine: hermes        
 ERROR  Invariant Violation: Failed to call into JavaScript module method AppRegistry.runApplication(). Module has not been registered as callable. Registered callable JavaScript modules (n = 11): Systrace, JSTimers, HeapCapture, SamplingProfiler, RCTLog, RCTDeviceEventEmitter, RCTNativeAppEventEmitter, GlobalPerformanceLogger, JSDevSupportModule, HMRClient, RCTEventEmitter.
        A frequent cause of the error is that the application entry file path is incorrect. This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native., js engine: hermes
